src/core/providers/ogr/qgsogrfeatureiterator.cpp

```diff
@@ -279,6 +277,23 @@ QgsOgrFeatureIterator::QgsOgrFeatureIterator( QgsOgrFeatureSource *source, bool
     std::sort( mRequestAttributes.begin(), mRequestAttributes.end() );
   }
 
+  // Install query logger
+#if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(3,7,0)
+  GDALDatasetSetQueryLoggerFunc( mConn->ds, [ ]( const char *pszSQL, const char *pszError, int64_t lNumRecords, int64_t lExecutionTimeMilliseconds, void *pQueryLoggerArg )
```

Comment:
for performance reasons, shoudn't this be installed only if SQL logging is enabled ?

Reply:
done